ampersand-dom-bindings.js

```diff
@@ -46,12 +51,22 @@ function switchHandler(binding, el, value) {
 }
 
 function getSelector(binding) {
-    if (typeof binding.selector === 'string') {
-        return binding.selector;
-    } else if (binding.hook) {
-        return '[data-hook~="' + binding.hook + '"]';
-    } else {
-        return '';
+    if(binding.firstMatchOnly !== undefined && binding.firstMatchOnly === true) {
+        if (typeof binding.selector === 'string') {
+            return {firstMatchOnly:binding.selector};
+        } else if (binding.hook) {
+            return {firstMatchOnly:'[data-hook~="' + binding.hook + '"]'};
+        } else {
+            return {firstMatchOnly:''};
+        }
+    }else{
+        if (typeof binding.selector === 'string') {
+          return binding.selector;
+        } else if (binding.hook) {
+          return '[data-hook~="' + binding.hook + '"]';
+        } else {
+          return '';
+        }
```

Comment:
Not the best implementation/solution, agreed on that. But the `getMatches` only receives the `binding.selector` (along with the element) passed through `getSelector`. So `getSelector` is the only one aware of all the `binding`'s props.

I wouldn't say _make me nervous_, but surely can be done better. 

What do you think? How can I solve this on `getMatches` (I assure you it was my first intention too, but got stuck when I saw it didn't get the `binding` on that function). Changing the `getMatches` function to accept a third parameter would require a full refactor of all the methods/cases calling it (every case uses it), so what you see here is more of a little hack/kludge than a full solution.